planet/src/main/java/com/google/planet/data/ItineraryGenerator.java

```diff
@@ -18,46 +18,98 @@
 import java.util.Arrays;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.stream.*;
+import com.google.maps.GeoApiContext;
+import com.google.maps.DistanceMatrixApi;
+import com.google.maps.GaeRequestHandler;
+import com.google.maps.model.DistanceMatrix;
+import com.google.maps.model.DistanceMatrixRow;
+import com.google.maps.model.DistanceMatrixElement;
+import com.google.maps.model.Duration;
 
 public final class ItineraryGenerator {
-    public List<ItineraryItem> generateItinerary(List<Event> events) { 
+    private String errorMessage;
+    
+    public Itinerary generateItinerary(List<Event> events) {
+        errorMessage = null; 
+        
         // Return an empty list if events are empty or only contains the starting point event
         if (events.size() <= 1) { 
-            return Arrays.asList();
+            errorMessage = "Itinerary can only be generated with at least one event!";
+            Itinerary itinerary = new Itinerary(Arrays.asList(), errorMessage);
```

Comment:
This looks like an invalid request, so the corresponding response code should be set accordingly. In this case, that's probably a 400

Reply:
fixed!